src/main/scala/ScalaModulePlugin.scala

```diff
@@ -147,6 +147,17 @@ object ScalaModulePlugin extends AutoPlugin {
 
   lazy val scalaModuleSettingsJVM: Seq[Setting[_]] = scalaModuleOsgiSettings
 
+  lazy val scalaModuleSettingsNative: Seq[Settings[_]] = Seq(
+    scalaVersion := "2.11.11",
+    skip in compile := System.getProperty("java.version").startsWith("1.6"),
+    test := {},
+    libraryDependencies := {
+      if (!scalaVersion.value.startsWith("2.11"))
+        libraryDependencies.value.filterNot(_.organization == "org.scala-native")
+      else libraryDependencies.value
+    }
```

Comment:
well, it actually makes sense, you just have explained it. That's how `scala-parser-combinators` was configured before I made any change to it (aggregating all the modules), also the compilation is driven by `sbt +compile` without strictly selecting a scala version with `++2.11.11` because this plugin _sets up the matix_.

Really, as I said, this comes mostly motivated by the initial PR referenced up there, might it be the case of only that project? probably, would it make sense to not aggregate modules in `scala-parser-combinators`? Maybe, it would make very cumbersome to work with it. I'm going to close this PR, feel free to continue the conversation in the original PR.

> you aggregate all the fooJVM/fooJS/fooNative in a single foo, and then **of course** foo/test will insist on running fooNative/test

That, in my view, it's a sign that there is something lacking in there. SBT gives the ability of cross-compiling to different Scala versions, `sbt-crossproject` attempts to give the ability of cross-compiling to different platforms. Aggregating projects is a common thing, workarounds like this one or like the one you mention (yeah, it's just another workaround) and not excuses to say _it's simple as that_ (oh yes, there are other workarounds, like creating _phantom_ `fooJS`, `fooJVM` modules, etc).